src/components/Auth/service.js

```diff
@@ -0,0 +1,61 @@
+const jwt = require('jsonwebtoken');
+const UserToken = require('./token_model');
+
+async function generateAccessToken(user) {
+    const payload = { userId: user.id };
+    const accessToken = jwt.sign(
+        payload,
+        'ACCESS_TOKEN_PRIVATE_KEY',
+        { expiresIn: '1m' },
+    );
+    return accessToken;
+}
+
+async function generateRefreshToken(user) {
+    const payload = { userId: user.id };
+    const refreshToken = jwt.sign(
+        payload,
+        'REFRESH_TOKEN_PRIVATE_KEY',
+        { expiresIn: '15m' },
+    );
+    const storedRefreshToken = await UserToken.findOne({ userId: user.id }).exec();
+    if (storedRefreshToken === null) {
+        await UserToken.create({ userId: user.id, token: refreshToken });
+    } else {
+        await UserToken.updateOne({ userId: user.id }, { token: refreshToken }).exec();
+    }
+    return refreshToken;
+}
+
+function verifyToken(req, res, next) {
+    try {
+        const token = req.headers.authorization.split(' ')[1];
+        const decoded = jwt.verify(token, 'ACCESS_TOKEN_PRIVATE_KEY');
+        req.userData = decoded;
+        next();
+    } catch (error) {
+        res.status(401).json({ status: false, message: 'Your session is not valid.', data: error });
+    }
+}
+
+async function verifyRefreshToken(req, res, next) {
+    const { token } = req.body;
+    if (token === null) return res.status(401).json({ status: false, message: 'Invalid request' });
```

Comment:
no need status key, see example of simple json response https://github.com/ChechaValerii/rest-api-response-format